internal/impl/gcp/processor_vertex_ai.go

```diff
@@ -0,0 +1,316 @@
+				Advanced().
+				Description("Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.").
+				Optional(),
+		)
+}
+
+func newVertexAIProcessor(conf *service.ParsedConfig, mgr *service.Resources) (p service.Processor, err error) {
+	ctx := context.Background()
+	proc := &vertexAIChatProcessor{}
+	var project string
+	project, err = conf.FieldString(vaiFieldProject)
+	if err != nil {
+		return
+	}
+	var location string
+	if conf.Contains(vaiFieldLocation) {
+		location, err = conf.FieldString(vaiFieldLocation)
+		if err != nil {
+			return
+		}
+	}
+	opts := []option.ClientOption{}
+	if conf.Contains(vaiFieldCredentialsJSON) {
+		var jsonFile string
+		jsonFile, err = conf.FieldString(vaiFieldCredentialsJSON)
+		if err != nil {
+			return
+		}
+		opts, err = getClientOptionWithCredential(jsonFile, opts)
+		if err != nil {
+			return
+		}
+	}
+	proc.client, err = genai.NewClient(ctx, project, location, opts...)
+	if err != nil {
+		return
+	}
+	defer func() {
+		if err != nil {
+			_ = proc.client.Close()
+		}
+	}()
+	proc.model, err = conf.FieldString(vaiFieldModel)
+	if err != nil {
+		return
+	}
+	if conf.Contains(vaiFieldPrompt) {
+		proc.userPrompt, err = conf.FieldInterpolatedString(vaiFieldPrompt)
+		if err != nil {
+			return
+		}
+	}
+	if conf.Contains(vaiFieldSystemPrompt) {
+		proc.systemPrompt, err = conf.FieldInterpolatedString(vaiFieldSystemPrompt)
+		if err != nil {
+			return
+		}
+	}
+	if conf.Contains(vaiFieldTemp) {
+		var temp float64
+		temp, err = conf.FieldFloat(vaiFieldTemp)
+		if err != nil {
+			return
+		}
+		proc.temp = genai.Ptr(float32(temp))
+	}
+	if conf.Contains(vaiFieldTopP) {
+		var topP float64
+		topP, err = conf.FieldFloat(vaiFieldTopP)
+		if err != nil {
+			return
+		}
+		proc.topP = genai.Ptr(float32(topP))
+	}
+	if conf.Contains(vaiFieldTopK) {
+		var topK int
+		topK, err = conf.FieldInt(vaiFieldTopK)
+		if err != nil {
+			return
+		}
+		proc.topK = genai.Ptr(int32(topK))
+	}
+	if conf.Contains(vaiFieldMaxTokens) {
+		var maxTokens int
+		maxTokens, err = conf.FieldInt(vaiFieldMaxTokens)
+		if err != nil {
+			return
+		}
+		proc.maxTokens = genai.Ptr(int32(maxTokens))
+	}
+	if conf.Contains(vaiFieldStop) {
+		proc.stopSequences, err = conf.FieldStringList(vaiFieldStop)
+		if err != nil {
+			return
+		}
+	}
+	if conf.Contains(vaiFieldPresencePenalty) {
+		var pp float64
+		pp, err = conf.FieldFloat(vaiFieldPresencePenalty)
+		if err != nil {
+			return
+		}
+		proc.presencePenalty = genai.Ptr(float32(pp))
+	}
+	if conf.Contains(vaiFieldFrequencyPenalty) {
+		var fp float64
+		fp, err = conf.FieldFloat(vaiFieldFrequencyPenalty)
+		if err != nil {
+			return
+		}
+		proc.frequencyPenalty = genai.Ptr(float32(fp))
+	}
+	var format string
+	format, err = conf.FieldString(vaiFieldResponseFormat)
+	if format == "json" {
+		proc.responseMIMEType = "application/json"
+	} else {
```

Comment:
Maybe return an error if the else branch is not `text`. Users could try to run the config with `--chilled` which I think lets them place any value they want in this field.

Reply:
Good catch!